### ISSUE TITLE
`auto` does not work on Mac OS X

### ISSUE THREAD

**Issue by philipbl** [pydoit/doit#49]
Adding the `auto` command line option does not work on Mac OS X (10.10.3). The task is run once, but then nothing occurs after that. I have played around with the source code and from what I can tell, the [process that `doit auto` creates](https://github.com/pydoit/doit/blob/master/doit/cmd_auto.py#L105-L107) is not exited when the file system change event is [received](https://github.com/pydoit/doit/blob/master/doit/cmd_auto.py#L93-L95).

Here is a simple `dodo.py` file that shows this problem:

```python
def task_test(): 
    return { 
        'actions': ['echo "test" > test.txt'], 
        'file_dep': ['watch.txt']             
        }
```

If you run `doit auto` and modify `watch.txt` on Mac OS X, nothing happens.

WIP at #101

**Comment by schettino72** [comment]
Thanks for the report. The problem is not so simple.

Check `filewatch.py`, it contains the notifier loop for both linux and mac.  From what I remember it works in a different way than the linux one. 
The linux version would start a new process after every notifcation but the Mac version doesnt.
I guess the difference came from the different API provided by the 2 systems. 

One advantage of creating a new process is that it doesnt cache the python imports.

I did not implement the Mac version myself... and I dont use Mac at all. So it would be great if you can contribute a patch.

**Comment by philipbl** [comment]
Sounds good. I will start working on it.

**Comment by schettino72** [comment]
Or maybe we should just use https://github.com/gorakhargosh/watchdog

I didnt look at it but I hope that using watchdog we could could support linux/mac/bsd/windows easily.

**Comment by philipbl** [comment]
I have a working version of doit that uses watchdog. I need to clean up the code before I submit a PR. Also, I only have access to Mac OS and Linux for testing. It would be great if we could test the new watchdog functionality on Windows and other OSes.

**Comment by schettino72** [comment]
fixed on #135
